packages/compass-import-export/src/components/import-file-input.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { useCallback, useMemo } from 'react';
+import {
+  FileInput,
+  createElectronFileInputBackend,
+} from '@mongodb-js/compass-components';
+
+type ImportFileInputProps = {
+  autoOpen?: boolean;
+  onCancel?: () => void;
+  selectImportFileName: (fileName: string) => void;
+  fileName: string;
+};
+
+function ImportFileInput({
+  autoOpen,
+  onCancel,
+  selectImportFileName,
+  fileName,
+}: ImportFileInputProps) {
+  const handleChooseFile = useCallback(
+    (files: string[]) => {
+      console.log('on change', files);
+      if (files.length > 0) {
+        selectImportFileName(files[0]);
+      } else if (typeof onCancel === 'function') {
+        onCancel();
+      }
+    },
+    [onCancel, selectImportFileName]
+  );
+
+  const backend = useMemo(() => {
+    // eslint-disable-next-line @typescript-eslint/consistent-type-imports
+    let electron: typeof import('@electron/remote');
```

Comment:
We should file a follow-up ticket for the whole project with what we need to do / change to get import and export to work in DE and this kind of thing should probably be mentioned in it.

Reply:
Actually thinking about it more, in the browser we'd have to keep a reference to the file object and not just the filename because we wouldn't be able to just read from the file with the fs module. So we would still need work here.

Reply:
Good thinking, I'll remove the try catch here so it explicitly fails in non-electron environments for now.